homeassistant/components/bmw_connected_drive/button.py

```diff
@@ -117,7 +119,10 @@ def __init__(
     async def async_press(self) -> None:
         """Press the button."""
         if self.entity_description.remote_function:
-            await self.entity_description.remote_function(self.vehicle)
+            try:
+                await self.entity_description.remote_function(self.vehicle)
+            except MyBMWAPIError as ex:
+                raise HomeAssistantError(str(ex)) from ex
```

Comment:
The `str(ex)` shouldn't be needed. Additionally, please add a descriptive end-user message in the HomeAssistantError message instead, as this will be shown in the UI and automation/script debuggers.

Reply:
The library exception messages are pretty descriptive, which is why I just repeated them using `str(ex)`.
Will remove the casting, though. 

Exception messages directly related to remote services:
- `f"Remote service failed with state '{status.state}'. Response: {status.details}"`
- `f"Did not receive remote service result for '{event_id}' in {_POLLING_TIMEOUT} seconds. Current state: {status.state.value}"`

General API errors are formatted as follows (with exception type always being a MyBMWAPIError or inhertied from it):
`{type(ex).__name__}: {_err["error"]} - {_err.get("error_description", "")}`

Please let me know if you need more descriptive exceptions for `HomeAssistantError`.